armory/scenarios/image_classification.py

```diff
@@ -41,10 +48,11 @@ def _evaluate(self, config: dict) -> dict:
                 split_type="train",
                 preprocessing_fn=preprocessing_fn,
             )
-            if config["defense"] is not None:
-                logger.info("loading defense")
-                defense = load_defense(config["defense"], classifier)
-                defense.fit_generator(train_data, **fit_kwargs)
+            if defense:
+                if defense_type in ["Trainer", "Transformer"]:
+                    logger.info(f"Training with {defense_type} defense...")
+                    defense = load_defense(config["defense"], classifier)
+                    defense.fit_generator(train_data, **fit_kwargs)
```

Comment:
This code won't work with DefensiveDistillation, the only concrete "Trainer" in ART.

Reply:
I believe we'll need to raise an issue with ART to support fit_generator

Reply:
I think that Transformer should probably go after the main training loop in its own space. For DefensiveDistillation, you need to first train the main network, and then train the distilled network.